01-shapefile-attributes.Rmd

```diff
@@ -254,39 +239,44 @@ footpath_HARV<-lines_HARV[lines_HARV$TYPE == "footpath",]
 footpath_HARV
 #how many features in our new object
 length(footpath_HARV)
+```
+
+Our subsetting operation reduces the `features` count from 13 to 2. This means
+that only two lines have the attribute "TYPE=footpath".
 
+We can also plot our subsetted shapefiles.
+```{r plot-subset-shapefile}
 #plot just footpaths
 plot(footpath_HARV,
+     col=lines_HARV$OBJECTID, #color set by "OBJECTID" attribute
      lwd=6,
      main="Footpaths at NEON Harvard Forest Field Site")
 ```
 
-Our subsetting operation reduces the `features` count from 13 to 2. This means
-that only two lines have the attribute "TYPE=footpath".
-
 ##Plot Lines by Attribute Value
-
-To plot vector data, colored by a set of attribute values, we can convert an
-attribute that has categorical data to class= `factor`. A factor is similar to a
-category - you can group vector objects by a particular category value - for
-example you can group all lines of `TYPE=footpath`. However in `R`, a factor
-can also have an *order*. 
+To plot vector data with the color determined by a set of attribute values,
+we must convert an attribute that is `categorical` data to class=
+`factor`. A **factor** is similar to a category - you can group vector objects
+by a particular category value - for example you can group all lines of
+`TYPE=footpath`. However, in `R`, a factor can also have a determined *order*. 
 
 If we convert the `lines_HARV$TYPE` column to a factor using `as.factor()`, then
 it will be much easier to plot features by attribute value.
 
-
 ```{r convert-to-factor }
 #view the original class of the TYPE column
```

Comment:
right - it depends upon how the data are imported. If the option for strings as factors is FALSE they won't be a factor. if it isn't set - it will be a factor. i suggest adding a line of clarification that you only have to perform those steps IF you have your R environmental set to by default turn off strings as factors. Some people do that, some don't. if i were teaching i'd skip that section assuming the room doesn't have that set but i often turn off strings as factors as do other folks so it's good to know how to do it.